docs/pages/enroll-resources/machine-id/deployment/bitbucket.mdx

```diff
@@ -0,0 +1,160 @@
+
+In order to allow your Pipelines workflow to authenticate with your Teleport
+cluster, you'll first need to create a join token. These tokens set out criteria
+by which the Auth Service decides whether or not to allow a bot or node to join.
+
+Create a file named `bot-token.yaml`, ensuring that you replace the
+`identity_provider_url`, `audience`, `workspace_uuid`, and `repository_uuid`
+with the values from Step 1.
+
+```yaml
+kind: token
+version: v2
+metadata:
+  name: example-bot
+spec:
+  roles: [Bot]
+  join_method: bitbucket
+  bot_name: example
+  bitbucket:
+    identity_provider_url: <Var name="identity-provider-url" />
+    audience: <Var name="audience" />
+    # allow specifies the rules by which the Auth Service determines if `tbot`
+    # should be allowed to join.
+    allow:
+    - workspace_uuid: <Var name="workspace-uuid" />
+      repository_uuid: <Var name="repository-uuid" />
+```
+
+Let's go over the token resource's fields in more detail:
+
+- `metadata.name` defines the name of the token. Note that this value will need
+  to be used in other parts of the configuration later.
+- `spec.bot_name` is the name of the Machine ID bot that this token will grant
+  access to. Note that this value will need to be used in other parts of the
+  configuration later.
+- `spec.roles` defines which roles that this token will grant access to. The
+  value of `[Bot]` states that this token grants access to a Machine ID bot.
+- `spec.join_method` defines the join method the token is applicable for. Since
+  this guide only focuses on Bitbucket Pipelines, you will set this to to
+  `bitbucket`.
+- `spec.bitbucket.identity_provider_url` is the identity provider URL shown in
+  the Bitbucket repository settings, under Pipelines and OpenID Connect.
+- `spec.bitbucket.audience` is the audience value shown in the Bitbucket
+  repository settings, under Pipelines and OpenID connect.
+- `spec.bitbucket.allow` is used to set rules for what Bitbucket Pipelines runs
+   will be able to authenticate by using the token.
+
+Refer to the [token reference](../../../reference/join-methods.mdx#bitbucket-pipelines-bitbucket)
+for a full list of valid fields.
+
+Apply this to your Teleport cluster using `tctl`:
+
+```code
+$ tctl create -f bot-token.yaml
+```
+
+## Step 4/4. Configure a Bitbucket Pipelines workflow
+
+With the bot and join token created, you can now configure a workflow that can
+authenticate to Teleport.
+
+This example workflow defines a "custom" pipeline that can be triggered manually
+from "Pipelines" or "Branches" views:
+
+```yaml
+image: atlassian/default-image:3
+
+pipelines:
+  custom:
+    run-tbot:
+      - step:
+          oidc: true
+          script:
+            # Download and extract Teleport
+            - wget https://cdn.teleport.dev/teleport-v(=teleport.version=)-linux-amd64-bin.tar.gz
+            - tar -xvf teleport-v(=teleport.version=)-linux-amd64-bin.tar.gz
+
+            # Run `tbot` in identity mode for SSH access
```

Comment:
We will have to change this if we backport bitbucket to v16, right?

Reply:
Good call, I'll tweak this in the backport to borrow step 5 from the other guides.